src/mapwindow.jl

```diff
@@ -273,13 +365,100 @@ end
 # This is slightly faster than a circular buffer
 @inline cyclecache(b, x) = b[1], (Base.tail(b)..., x)
 
+
 replace_function(f) = f
-replace_function(::typeof(median!)) = function(v)
-    inds = indices(v,1)
-    Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+
+replace_function(::typeof(median!)) = function(v,m_histogram,mode,window)
+   # Handle the boundary points with mode -1
+    if(mode==-1)
+        inds = indices(v,1)
+        return Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+    else
+        
+        window_size=size(v,1)
+        dims = map(x->x.stop-x.start+1,window)
```

Comment:
Better to use the "official interface" of ranges rather than accessing fields directly---use `first(x)` and `last(x)` rather than `x.start` and `x.stop`. (If someone renames the fields or redefines the implementation, they can't deprecate your direct access of the fields whereas they can make sure that `first` and `last` continue to work properly. This actually happened not so long ago with `LinSpace` and `StepRangeLen`.)

However, in this particular case I think this is equivalent to `map(length, window)`, and that's much clearer for readers of your code.